libs/domain/user/registration/registration.component.ts

```diff
@@ -0,0 +1,217 @@
+
+  protected genderService = resolve(GenderService);
+  protected genders = signal(this.genderService.get());
+
+  protected onSubmit(event: Event): void {
+    event.preventDefault();
+
+    if (this.validateForm()) {
+      this.registerUser();
+    }
+  }
+
+  protected validateForm(): boolean {
+    this.setCheckboxValidity();
+    this.setPasswordValidity();
+    return !!(
+      this.form?.checkValidity() &&
+      this.acceptedTerms?.checked &&
+      this.passwordComponent?.isValid()
+    );
+  }
+
+  protected setCheckboxValidity(): void {
+    this.hasAgreementError = !this.acceptedTerms?.checked;
+  }
+
+  protected setPasswordValidity(): void {
+    this.hasPasswordError = !this.passwordComponent?.isValid();
+  }
+
+  protected async registerUser(): Promise<void> {
+    this.isLoading = true;
+
+    try {
+      await firstValueFrom(
+        this.registrationService.register(this.getFormData())
+      );
+      this.hasGenericError = false;
+    } catch (e) {
+      this.hasGenericError = true;
+    } finally {
+      this.isLoading = false;
+    }
+  }
+
+  protected override render(): TemplateResult {
+    return html`
+      ${when(
+        this.hasGenericError,
+        () => html`
+          <oryx-notification type="error" scheme="dark">
+            ${this.i18n('user.registration.something-went-wrong')}.
+          </oryx-notification>
+        `
+      )}
+
+      <form @submit=${this.onSubmit}>
+        ${this.fieldRenderer.buildForm(this.getFields())}
+        <oryx-password-input
+          .strategy="${this.$options()?.passwordVisibility}"
+          .minLength=${this.$options()?.minLength}
+          .maxLength=${this.$options()?.maxLength}
+          .minUppercaseChars=${this.$options()?.minUppercaseChars}
+          .minNumbers=${this.$options()?.minNumbers}
+          .minSpecialChars=${this.$options()?.minSpecialChars}
+          .label=${this.i18n('login.password')}
+          .errorMessage=${this.hasPasswordError &&
+          this.i18n(
+            'user.registration.password-doesn’t-satisfy-all-the-criteria'
+          )}
+          required
+          @input=${() => (this.hasPasswordError = false)}
+        >
+          <input type="password" name="password" required />
+        </oryx-password-input>
+        <div class="agreement">
+          <oryx-checkbox ?hasError=${this.hasAgreementError}>
+            <input
+              type="checkbox"
+              name="acceptedTerms"
+              required
+              @change=${this.hasAgreementError && this.setCheckboxValidity}
+            />
+          </oryx-checkbox>
+          <span>
+            ${this.i18n('user.registration.i-accept-the-<terms>', {
+              terms: {
+                value: 'terms and conditions',
+                link: {
+                  href: this.$options()?.termsAndConditionsLink || '',
+                  target: '_blank',
+                },
+              },
+            })}
+          </span>
+        </div>
```

Comment:
we should add text to the checkbox component rather than adding separate span; the span text should be clickable as well. We should stil be able to click on the link though.